pkg/kubectl/cmd/attach.go

```diff
@@ -116,12 +122,37 @@ type AttachOptions struct {
 // Complete verifies command line arguments and loads data from the command environment
 func (p *AttachOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, argsIn []string) error {
 	if len(argsIn) == 0 {
-		return cmdutil.UsageError(cmd, "POD is required for attach")
+		return cmdutil.UsageError(cmd, "at least one argument is required for attach")
+	}
+	if len(argsIn) > 2 {
+		return cmdutil.UsageError(cmd, fmt.Sprintf("expected fewer than three arguments: POD or TYPE/NAME or TYPE NAME, saw %d: %s", len(argsIn), argsIn))
 	}
-	if len(argsIn) > 1 {
-		return cmdutil.UsageError(cmd, fmt.Sprintf("expected a single argument: POD, saw %d: %s", len(argsIn), argsIn))
+
+	cmdNamespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	if strings.Contains(argsIn[0], "/") || len(argsIn) == 2 {
```

Comment:
Yeah, I know both `ResourceNames` and `ResourceTypeOrNameArgs` inside have a func named `splitResourceTypeName` can do the parsing, but the logic here is for only one argument, it could be only a pod name eg:nginx or rs name rs/nginx if it's the first one, we assume it must be a pod name, so we only need to call p.PodName = argsIn[0]. if it's second one, it may not be a pod's name, so we call ResourceTypeOrNameArgs to get resource object instead.
Also `ResourceNames` fisrt parameter should be a given type we already know before we call it, i think call `ResourceTypeOrNameArgs` is more suitable if we don't know whether it's a rs or pod or statefulSet etc... 
Is there something i misunderstood ?